### PR TITLE
Budi added a cool chart

### PR DESCRIPTION
**Pull request by RazvanBudacaTechquarter**
Pressing the button will call the API to load and draw a chart with the most important words of the text.

For now it's all static data. To convert, we just call fetch to our server and use the list from the response to get info about our entities.

### CODE REVIEW

File: src/main/webapp/src/components/songInfo.js [hunk 1]
```diff
@@ -144,6 +146,32 @@ class SongInfo extends React.Component {
...
+    function drawChart() {
+      const data = new google.visualization.DataTable();
+      data.addColumn('string', 'Name');
+      data.addColumn('number', 'Salience');
+      songInfo.entityAnalysis.forEach((entity) => {
+        data.addRow([entity.word, entity.salience]);
+      });
+
+      const options = {
+        'title': 'Most important words',
```
Comment (jeinygroove) [inline src/main/webapp/src/components/songInfo.js]: you can remove `'` around `title`

File: src/main/webapp/src/components/songInfo.js [hunk 2]
```diff
@@ -144,6 +146,32 @@ class SongInfo extends React.Component {
...
+        'width':600,
```
Comment (jeinygroove) [inline src/main/webapp/src/components/songInfo.js]: You can remove `'` around `width`. And add space after `:`

File: src/main/webapp/src/components/songInfo.js [hunk 3]
```diff
@@ -144,6 +146,32 @@ class SongInfo extends React.Component {
...
+        'height':500
```
Comment (jeinygroove) [inline src/main/webapp/src/components/songInfo.js]: You can remove `'` around `height`. And add space after `:`

File: src/main/webapp/src/components/songInfo.js [hunk 4]
```diff
@@ -163,17 +191,8 @@ class SongInfo extends React.Component {
...
-                  <ListItem key={index + 1}>
-                    {wordInfo.word +
-                      " - " +
-                      wordInfo.salience +
-                      " - " +
-                      wordInfo.category}
-                  </ListItem>
-                ))}
-              </List>
+              <button onClick={loadChartAPI}>Get Entity Analysis results</button>
```
Comment (jeinygroove) [inline src/main/webapp/src/components/songInfo.js]: Of course we don't want force user to push some button just to see the charts. One of the best solution is to use `ref`.

File: src/main/webapp/src/components/songInfo.js [hunk 5]
```diff
@@ -163,17 +191,8 @@ class SongInfo extends React.Component {
...
+              <div id="chart-container"></div>
```
Comment (jeinygroove) [inline src/main/webapp/src/components/songInfo.js]: Add `ref={() => loadChartAPI()}` and remove button that you had, so this function will be called after rendering.

More info [here](https://reactjs.org/docs/refs-and-the-dom.html)

File: src/main/webapp/src/components/songInfo.js (hunk 4 above)
Comment (RazvanBudacaTechquarter) [reply src/main/webapp/src/components/songInfo.js]: Oh ok I thought we wanted a button. My bad.

**Review by jeinygroove — COMMENTED** [review]
I also found a better solution with `ref`. In my first suggestion you still get an element with `getElementById`, but `ref` provides you an opportunity to get this element immediately. 

For this you need to add `ref={(elem) => {
                  if (elem) loadChartAPI(elem);
                }}`
Short explanation: we need to check if `elem` is not null, because ref is called twice, one time with null and one time with loaded element.

Then you need to add `element` argument to `loadChartAPI` and `drawChart` functions.

Then you need to replace `google.charts.setOnLoadCallback(drawChart);` with `google.charts.setOnLoadCallback(function() { drawChart(element) });`.  (You can't replace `function() { drawChart(element) }` with `drawChart(element)`, it's quite long explanation, I can tell it if you'll ask or I'll add it to this comment a little bit later).

And the main part: `const entityChart = new google.visualization.PieChart(element);` instead of calling `getElementById`

File: src/main/webapp/src/components/songInfo.js [hunk 6]
```diff
@@ -144,6 +146,32 @@ class SongInfo extends React.Component {
...
+    function drawChart() {
+      const data = new google.visualization.DataTable();
+      data.addColumn('string', 'Name');
+      data.addColumn('number', 'Salience');
+      songInfo.entityAnalysis.forEach((entity) => {
+        data.addRow([entity.word, entity.salience]);
+      });
+
+      const options = {
+        title : 'Most important words',
```
Comment (jeinygroove) [inline src/main/webapp/src/components/songInfo.js]: remove space between `title` and `:`. Same for other fields.

Comment (RazvanBudacaTechquarter) [reply src/main/webapp/src/components/songInfo.js]: Spaces are my enemy.

**Comment by RazvanBudacaTechquarter** [comment]
> I also found a better solution with `ref`. In my first suggestion you still get an element with `getElementById`, but `ref` provides you an opportunity to get this element immediately.
> 
> For this you need to add `ref={(elem) => { if (elem) loadChartAPI(elem); }}`
> Short explanation: we need to check if `elem` is not null, because ref is called twice, one time with null and one time with loaded element.
> 
> Then you need to add `element` argument to `loadChartAPI` and `drawChart` functions.
> 
> Then you need to replace `google.charts.setOnLoadCallback(drawChart);` with `google.charts.setOnLoadCallback(function() { drawChart(element) });`. (You can't replace `function() { drawChart(element) }` with `drawChart(element)`, it's quite long explanation, I can tell it if you'll ask or I'll add it to this comment a little bit later).
> 
> And the main part: `const entityChart = new google.visualization.PieChart(element);` instead of calling `getElementById`

I didn't quite get why this method is better in our case. Can you tell me again?

**Comment by jeinygroove** [comment]
> > I also found a better solution with `ref`. In my first suggestion you still get an element with `getElementById`, but `ref` provides you an opportunity to get this element immediately.
> > For this you need to add `ref={(elem) => { if (elem) loadChartAPI(elem); }}`
> > Short explanation: we need to check if `elem` is not null, because ref is called twice, one time with null and one time with loaded element.
> > Then you need to add `element` argument to `loadChartAPI` and `drawChart` functions.
> > Then you need to replace `google.charts.setOnLoadCallback(drawChart);` with `google.charts.setOnLoadCallback(function() { drawChart(element) });`. (You can't replace `function() { drawChart(element) }` with `drawChart(element)`, it's quite long explanation, I can tell it if you'll ask or I'll add it to this comment a little bit later).
> > And the main part: `const entityChart = new google.visualization.PieChart(element);` instead of calling `getElementById`
> 
> I didn't quite get why this method is better in our case. Can you tell me again?

Because in this case you don't need to add `id` to the element and you don't need to get this element manually when you already "have" it. I don't want to say that this is better, it's a **little bit** better  because you won't depend on `id` value.

This also shows `ref`'s logic, we don't just use it to call a function after loading of the element, but actually **refer** to this element.

**Comment by RazvanBudacaTechquarter** [comment]
> > > I also found a better solution with `ref`. In my first suggestion you still get an element with `getElementById`, but `ref` provides you an opportunity to get this element immediately.
> > > For this you need to add `ref={(elem) => { if (elem) loadChartAPI(elem); }}`
> > > Short explanation: we need to check if `elem` is not null, because ref is called twice, one time with null and one time with loaded element.
> > > Then you need to add `element` argument to `loadChartAPI` and `drawChart` functions.
> > > Then you need to replace `google.charts.setOnLoadCallback(drawChart);` with `google.charts.setOnLoadCallback(function() { drawChart(element) });`. (You can't replace `function() { drawChart(element) }` with `drawChart(element)`, it's quite long explanation, I can tell it if you'll ask or I'll add it to this comment a little bit later).
> > > And the main part: `const entityChart = new google.visualization.PieChart(element);` instead of calling `getElementById`
> > 
> > 
> > I didn't quite get why this method is better in our case. Can you tell me again?
> 
> Because in this case you don't need to add `id` to the element and you don't need to get this element manually when you already "have" it. I don't want to say that this is better, it's a **little bit** better because you won't depend on `id` value.
> 
> This also shows `ref`'s logic, we don't just use it to call a function after loading of the element, but actually **refer** to this element.

Sounds interesting, imma go for it.

**Review by jeinygroove — APPROVED** [review]

File: src/main/webapp/src/components/songInfo.js [hunk 7]
```diff
@@ -144,7 +146,32 @@ class SongInfo extends React.Component {
       ],
     };
 
-    // TODO display entities with charts
+    function drawChart(element) {
```
Comment (amaurymedeiros) [inline src/main/webapp/src/components/songInfo.js]: `drawChart` is a generic name. Please use a more meaningful one and/or add some docs to explain which chart we're drawing here (probably an entity pie chart) and giving details about how it's expected to look like

**Comment by RazvanBudacaTechquarter** [comment]
Thank you Olya, very cool!

**Review by amaurymedeiros — COMMENTED** [review]
LGTM

**Pull request merged**